packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -3,13 +3,66 @@ import type * as MessageTypes from '@trezor/blockchain-link-types/lib/messages';
 import { CustomError } from '@trezor/blockchain-link-types/lib/constants/errors';
 import { BaseWorker, ContextType, CONTEXT } from '../baseWorker';
 import { MESSAGES, RESPONSES } from '@trezor/blockchain-link-types/lib/constants';
-import { Connection, PublicKey } from '@solana/web3.js';
+import { Connection, ParsedTransactionWithMeta, PublicKey } from '@solana/web3.js';
+import { Transaction } from '../..';
+import { solanaUtils } from '@trezor/blockchain-link-utils';
 
 export type SolanaAPI = Connection;
 
 type Context = ContextType<SolanaAPI>;
 type Request<T> = T & Context;
 
+const getAllSignatures = async (
+    api: SolanaAPI,
+    descriptor: MessageTypes.GetAccountInfo['payload']['descriptor'],
+) => {
+    let lastSignature: string | undefined;
+    let keepFetching = true;
+    let allSignatures: string[] = [];
+
+    const limit = 100;
+    while (keepFetching) {
+        const signaturesInfos = // eslint-disable-next-line no-await-in-loop
+            await api.getSignaturesForAddress(new PublicKey(descriptor), {
+                before: lastSignature,
+                limit,
+            });
+
+        const signatures = signaturesInfos.map(info => info.signature);
+        lastSignature = signatures[signatures.length - 1];
+        keepFetching = signatures.length === limit;
```

Comment:
equality checks in unbounded loops aren't a good idea overall. What if you fetch 101 signatures? 
I would change this to `signatures.length < limit`

Reply:
In the end I decided to keep this as is.. `signatures.length < limit` is not right.. negated value of this is the correct solution. If for some reason we fetch more than 100, loop will end,..if fewer, the loop will end...that makes sense to me.